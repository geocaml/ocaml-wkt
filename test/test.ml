open Bytesrw

let print filename =
  In_channel.with_open_bin filename @@ fun ic ->
  In_channel.fold_lines (fun () l ->
    let reader = Bytes.Reader.of_string l in 
    let wkt = Wkt.Geometry.decode reader in
    Fmt.pr "%a\n%!" Wkt.Geometry.pp wkt
  ) () ic

let print_crs filename =
  In_channel.with_open_bin filename @@ fun ic ->
  let reader = Bytes.Reader.of_in_channel ic in
  let crs = Wkt.Crs.decode ~filename reader in
  Fmt.pr "%a\n%!" Wkt.Crs.pp crs

let () =
  print "test.wkt";
  print_crs "./EPSG-CRS-3300.wkt"
