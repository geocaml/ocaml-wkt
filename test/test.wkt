point (10 10)
point m (10 10 10)
point zm (10 10 10 1234)
linestring (99 99,   100   100, 111 111 )
linestring zm (99 99 23 22,   100   100 10 10, 2 11 2  111 )
Polygon ((10 10, 10 20, 20 20, 20 15, 10 10))
Polygon ((10 10, 20 15, 10 10), (10 10, 20 30, 40 40))      
