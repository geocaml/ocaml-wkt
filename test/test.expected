point (Nx Info:
         Shape: [2]
         Dtype: float64
         Strides: [1]
         Offset: 0
         Size: 2
         Data: [10, 10]
        (2D))
point (Nx Info:
         Shape: [3]
         Dtype: float64
         Strides: [1]
         Offset: 0
         Size: 3
         Data: [10, 10, 10]
        (2D-m))
point (Nx Info:
         Shape: [4]
         Dtype: float64
         Strides: [1]
         Offset: 0
         Size: 4
         Data: [10, 10, 10, 1234]
        (3D-m))
linestring (Nx Info:
              Shape: [3x2]
              Dtype: float64
              Strides: [2; 1]
              Offset: 0
              Size: 6
              Data: [[99, 99],
                     [100, 100],
                     [111, 111]]
             (2D))
linestring (Nx Info:
              Shape: [3x4]
              Dtype: float64
              Strides: [4; 1]
              Offset: 0
              Size: 12
              Data: [[99, 99, 23, 22],
                     [100, 100, 10, 10],
                     [2, 11, 2, 111]]
             (3D-m))
polygon (Nx Info:
           Shape: [5x2]
           Dtype: float64
           Strides: [2; 1]
           Offset: 0
           Size: 10
           Data: [[10, 10],
                  [10, 20],
                  [20, 20],
                  [20, 15],
                  [10, 10]]
          (2D))
polygon (Nx Info:
           Shape: [3x2]
           Dtype: float64
           Strides: [2; 1]
           Offset: 0
           Size: 6
           Data: [[10, 10],
                  [20, 15],
                  [10, 10]]
         
         Nx Info:
           Shape: [3x2]
           Dtype: float64
           Strides: [2; 1]
           Offset: 0
           Size: 6
           Data: [[10, 10],
                  [20, 30],
                  [40, 40]]
          (2D))
